cloudweatherreport/utils.py

```diff
@@ -308,3 +293,25 @@ def juju_cmd(cmd):
     cmd = cmd.split() if isinstance(cmd, str) else cmd
     cmd = ['juju'] + cmd
     return subprocess.check_output(cmd).decode('utf-8')
+
+
+def guess_provider_name(controller_name):
+    """Guess provider name from the controller name.
+
+    If connection to the juju_client fails, no way to know the actual cloud
+    provider name. This can be used to guess the provider name.
+    :param controller_name: Controller name
+    """
+    controller_name = controller_name.lower()
+    if 'gce' in controller_name or 'google' in controller_name:
+        return 'gce'
+    if 'azure' in controller_name:
+        return 'azure'
+    if 'maas' in controller_name:
+        return 'maas'
+    if 'openstack' in controller_name:
+        return 'openstack'
+    if ('lxc' in controller_name or 'local' in controller_name or
+            'lxc' in controller_name):
+        return 'lxd'
+    return 'aws'
```

Comment:
Is defaulting to aws really the right thing here? I'd be tempted to just say "unknown", as that is the truth.